profiles/views_test.py

```diff
@@ -292,3 +280,62 @@ def test_staff_sees_entire_profile(self):
         self.client.force_login(self.user1)
         resp = self.client.get(self.url2)
         assert resp.json() == ProfileSerializer().to_representation(profile)
+
+
+class ProfilePATCHTests(ProfileBaseTests):
+    """
+    Tests for profile PATCH
+    """
+    def test_patch_own_profile(self):
+        """
+        A user PATCHes their own profile
+        """
+        with mute_signals(post_save):
+            ProfileFactory.create(user=self.user1, filled_out=False, agreed_to_terms_of_service=False)
+        self.client.force_login(self.user1)
+
+        with mute_signals(post_save):
+            new_profile = ProfileFactory.create(filled_out=False)
+        new_profile.user.social_auth.create(
+            provider=EdxOrgOAuth2.name,
+            uid="{}_edx".format(new_profile.user.username)
+        )
+        patch_data = ProfileSerializer().to_representation(new_profile)
+
+        resp = self.client.patch(self.url1, content_type="application/json", data=json.dumps(patch_data))
+        assert resp.status_code == 200
+        assert self.returned_serializer is ProfileSerializer
```

Comment:
these test for the serializer used are a little convoluted, and in this case, sort of shoe-horned into a case that is testing something else. i think we should get rid of the monkey patch and create test cases that focus directly on the serializer use. you could start with something like:

```python
class ProfileSerializerTests(ProfileBaseTests):
    def set_profile_and_login(self, **profile_kwargs):
        with mute_signals(post_save):
            ProfileFactory.create(user=self.user1, **profile_kwargs)
        self.client.force_login(self.user1)
    
    @patch('profiles.serializers.ProfileSerializer', autospec=True)
    def test_not_filled_out_serialization(self, mock_serializer):
        self.set_profile_and_login(self, filled_out=False)
        self.client.patch(self.url1, content_type="application/json", data=json.dumps(patch_data))
        assert mock_serializer.called
```
then add tests for the other conditions. the monkey-patching adds confusion, and the returned_serializer assertion in this test case sort of obscures its intention and makes it less readable